### ISSUE TITLE
Filtering over nested object

### ISSUE THREAD

**Issue by macher91** [barseghyanartur/django-elasticsearch-dsl-drf#57]
Hi,
I have field:
```
provider = fields.NestedField(
        properties={
            'id': fields.IntegerField(),
            'name': fields.TextField(boost=0.8)
        }
    )
```
In views I have:
```
filter_fields = {
        'provider_id': 'provider.id'
    }
```
When I filtering over provider_id it's genereting:
```
{
  "query": {
    "bool": {
      "must": [{"match": {"provider.id": "1"}}]
    }
  }
}
```
It always return 0 elements.

How to achive to get filtered elements?

**Comment by barseghyanartur** [comment]
@macher91:

An entire secton of the documentation is dedicated to nested fields and filtering.

http://django-elasticsearch-dsl-drf.readthedocs.io/en/0.8.2/nested_fields_usage_examples.html

**Comment by supadrupa** [comment]
I have the same problem

**Comment by barseghyanartur** [comment]
@supadrupa:

Did you read the docs? What didn't work?

**Comment by supadrupa** [comment]
documents.py 
```
@organization.doc_type
class ProjectOrganizationDocument(DocType):
    project = fields.NestedField(
        properties={
            'id': fields.IntegerField(),
            'name': fields.TextField(
                fields={
                    'raw': KeywordField(),
                }
            ),
            'site_name': fields.TextField(
                fields={
                    'raw': KeywordField(),
                }
            )
        }
    )
    name = KeywordField()
    organization_id = KeywordField()
    class Meta:
        model = ProjectOrganization
```

views.py
```
class ProjectOrganizationViewSet(DocumentViewSet):
    document = ProjectOrganizationDocument
    serializer_class = ProjectOrganizationSerializer
    permission_classes = (IsAuthenticatedOrReadOnly,)
    filter_backends = (
        FilteringFilterBackend,
    )
    filter_fields = {
        'project_id': 'project.id',
        'name': 'name'
    }
```

filter project_id does not work
http://127.0.0.1:8001/api/project_organization/?project_id=1
[]

**Comment by supadrupa** [comment]
```
    {
        "organization_id": "b396bf24-2d83-4762-9e5b-2883a41bb860",
        "name": "Автосервис Ника",
        "project": [
            {
                "id": 1,
                "name": "Справочный сайт spravbiz.ru",
                "site_name": "spravbiz.ru"
            }
        ]
    }
```
data without filter
http://127.0.0.1:8001/api/project_organization/

**Comment by barseghyanartur** [comment]
@supadrupa:

Thanks for the examples. Could you post the actual Elastic query being fired when calling the `http://127.0.0.1:8001/api/project_organization/?project_id=1`?

**Comment by macher91** [comment]
I post here how it's look in my case. So propobly the elastic query from @supadrupa will be:

```
{
  "query": {
    "bool": {
      "must": [{"match": {"project.id": "1"}}]
    }
  }
}
```

**Comment by barseghyanartur** [comment]
@macher91, @supadrupa:

It's currently being worked on. Somewhere this week initial support for nested filtering (and as a beginning of such, examples of how to have nested facets) will be released.

**Comment by barseghyanartur** [comment]
@macher91, @supadrupa:

Solved in version 0.8.4. See the docs link (given above) for detailed explanations.

In short:

If we have this kind of a mapping:

```python
{
    'continent': {
        'id': 2,
        'name': 'Asia',
        'country': {
            'id': 3,
            'name': 'Netherlands',
            'city': {
                'id': 5,
                'name': 'Amsterdam',
            }
        }
    }
}
```

Then the correspondent way of nested filters look as shown below. Note, that we use ``NestedFilteringFilterBackend`` and ``nested_filter_fields``.

```python
from django_elasticsearch_dsl_drf.filter_backends import (
    DefaultOrderingFilterBackend,
    FacetedSearchFilterBackend,
    FilteringFilterBackend,
    GeoSpatialFilteringFilterBackend,
    GeoSpatialOrderingFilterBackend,
    NestedFilteringFilterBackend,
    OrderingFilterBackend,
    SearchFilterBackend,
    SuggesterFilterBackend,
)
from django_elasticsearch_dsl_drf.pagination import LimitOffsetPagination
from django_elasticsearch_dsl_drf.viewsets import DocumentViewSet

class AddressDocumentViewSet(DocumentViewSet):

    document = AddressDocument
    serializer_class = AddressDocumentSerializer
    lookup_field = 'id'
    filter_backends = [
        FacetedSearchFilterBackend,
        FilteringFilterBackend,
        OrderingFilterBackend,
        SearchFilterBackend,
        GeoSpatialFilteringFilterBackend,
        GeoSpatialOrderingFilterBackend,
        NestedFilteringFilterBackend,
        DefaultOrderingFilterBackend,
        SuggesterFilterBackend,
    ]
    pagination_class = LimitOffsetPagination
    # Define search fields
    search_fields = (
        'street',
        'zip_code',
        'city.name',
        'city.country.name',
    )
    # Define filtering fields
    filter_fields = {
        'id': None,
        'city': 'city.name.raw',
        'country': 'city.country.name.raw',
    }
    # Nested filtering fields
    nested_filter_fields = {
        'continent_country': {
            'field': 'continent.country.name.raw',
            'path': 'continent.country',
        },
        'continent_country_city': {
            'field': 'continent.country.city.name.raw',
            'path': 'continent.country.city',
        },
        'continent_country_city_id': {
            'field': 'continent.country.city.id',
            'path': 'continent.country.city',
        },
    }
    # Define geo-spatial filtering fields
    geo_spatial_filter_fields = {
        'location': {
            'lookups': [
                LOOKUP_FILTER_GEO_BOUNDING_BOX,
                LOOKUP_FILTER_GEO_DISTANCE,
                LOOKUP_FILTER_GEO_POLYGON,

            ],
        },
    }
    # Define ordering fields
    ordering_fields = {
        'id': None,
        'street': None,
        'city': 'city.name.raw',
        'country': 'city.country.name.raw',
        'zip_code': None,
    }
    # Define ordering fields
    geo_spatial_ordering_fields = {
        'location': None,
    }
    # Specify default ordering
    ordering = (
        'id',
        'street.raw',
        'city.name.raw',
    )
    # Suggester fields
    suggester_fields = {
        'street_suggest': {
            'field': 'street.suggest',
            'suggesters': [
                SUGGESTER_COMPLETION,
            ],
        },
        'city_suggest': {
            'field': 'city.name.suggest',
            'suggesters': [
                SUGGESTER_COMPLETION,
            ],
        },
        'country_suggest': {
            'field': 'city.country.name.suggest',
            'suggesters': [
                SUGGESTER_COMPLETION,
            ],
        }
    }

    # Facets
    faceted_search_fields = {
        'city': {
            'field': 'city.name.raw',
            'enabled': True,
        },
        'country': {
            'field': 'city.country.name.raw',
            'enabled': True,
        },
    }
```
